emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java

```diff
@@ -97,9 +97,14 @@ private void updateKeyboardState() {
     rect = Utils.windowVisibleDisplayFrame(context);
 
     final int availableHeight = rootView.getHeight() - viewInset - (!Utils.isFullScreen(context) ? rect.top : 0);
-    final int keyboardHeight = Utils.shouldOverrideRegularCondition(context, editText) ?
-        Utils.getScreenHeight(context) - editText.getBottom() - (!Utils.isFullScreen(context) ? rect.top : 0)
-        : availableHeight - (rect.bottom - rect.top);
+
+    int keyboardHeight;
+
+    if (Utils.shouldOverrideRegularCondition(context, editText)) {
+      keyboardHeight = (int) ((availableHeight - editText.getBottom() - (0.252 * availableHeight)) + (!Utils.isFullScreen(context) ? availableHeight * 0.055 : 0));
```

Comment:
Can you explain what is going on here?

Reply:
It is a magic calculation based off of some fractions I've found in the Android source code. Not SUPER reliable, but it seemed to work on all the phones I've had. Can you show me a screenshot on how it looks like on yours?